samples-generator/lib/createPointCloudTile.js

```diff
@@ -607,60 +689,111 @@ function getBatchIds(batchIds) {
         for (i = 0; i < pointsLength; ++i) {
+            minComp[0] = Math.min(minComp[0], batchIds[i]);
+            maxComp[0] = Math.max(maxComp[0], batchIds[i]);
             buffer.writeUInt16LE(batchIds[i], i * sizeOfUint16);
         }
         componentType = 'UNSIGNED_SHORT';
     } else {
-        buffer = Buffer.alloc(pointsLength * sizeOfUint32);
+        buffer = Buffer.alloc(pointsLength * sizeOfUint8);
         for (i = 0; i < pointsLength; ++i) {
-            buffer.writeUInt32LE(batchIds[i], i * sizeOfUint32);
+            buffer.writeUInt8(batchIds[i], i * sizeOfUint8);
+            minComp[0] = Math.min(minComp[0], batchIds[i]);
+            maxComp[0] = Math.max(maxComp[0], batchIds[i]);
         }
-        componentType = 'UNSIGNED_INT';
+        componentType = 'UNSIGNED_BYTE';
     }
 
     return {
         buffer : buffer,
-        propertyName : 'BATCH_ID',
+        propertyName : use3dTilesNext ? '_BATCH_ID0' : 'BATCH_ID',
         componentType : componentType,
         type : 'SCALAR',
-        batchLength : batchLength
+        batchLength : batchLength,
+        min : minComp,
+        max : maxComp,
+        count : pointsLength
     };
 }
 
-function getColorsRGB(colors) {
+/**
+ *
+ * @param {Array<Object>} colors List of colors to use
+ * @param {Boolean} use3dTilesNext Forces RGBA mode due to byte alignment
+ *                                 requirements in glTF.
+ */
+function getColorsRGB(colors, use3dTilesNext) {
     var colorsLength = colors.length;
-    var buffer = Buffer.alloc(colorsLength * 3);
+    var multiple = use3dTilesNext ? 4 : 3;
+    var buffer = Buffer.alloc(colorsLength * multiple);
+
+    var minComp = new Array(multiple).fill(Number.POSITIVE_INFINITY);
+    var maxComp = new Array(multiple).fill(Number.NEGATIVE_INFINITY);
+
+    // avoid performing min max calculation on padding
+    if (use3dTilesNext) {
+        minComp[3] = 0;
+        maxComp[3] = 0;
+    }
+
     for (var i = 0; i < colorsLength; ++i) {
         var color = colors[i];
         var r = Math.floor(color.red * 255);
         var g = Math.floor(color.green * 255);
         var b = Math.floor(color.blue * 255);
-        buffer.writeUInt8(r, i * 3);
-        buffer.writeUInt8(g, i * 3 + 1);
-        buffer.writeUInt8(b, i * 3 + 2);
+
+        buffer.writeUInt8(r, i * multiple);
+        buffer.writeUInt8(g, i * multiple + 1);
+        buffer.writeUInt8(b, i * multiple + 2);
+
+        if (use3dTilesNext) {
+            buffer.writeUInt8(0, i * multiple + 3);
+        }
+
+        minComp[0] = Math.min(minComp[0], r);
+        minComp[1] = Math.min(minComp[1], g);
+        minComp[2] = Math.min(minComp[2], b);
+        maxComp[0] = Math.max(maxComp[0], r);
+        maxComp[1] = Math.max(maxComp[1], g);
+        maxComp[2] = Math.max(maxComp[2], b);
+
     }
     return {
         buffer : buffer,
-        propertyName : 'RGB',
+        propertyName : use3dTilesNext ? '_RGB' : 'RGB',
```

Comment:
The accessor type for rgb colors should be a vec3 instead of a vec4. The min and max arrays should be 3 elements each as well.

From `PointCloudsRGB`:

```js
    {
      "bufferView": 3,
      "byteOffset": 0,
      "componentType": 5121,
      "type": "VEC4",
      "count": 1000,
      "min": [
        0,
        0,
        0,
        0
      ],
      "max": [
        254,
        254,
        254,
        0
      ]
    }
```

Should be

```js
    {
      "bufferView": 3,
      "byteOffset": 0,
      "componentType": 5121,
      "type": "VEC3",
      "count": 1000,
      "min": [
        0,
        0,
        0
      ],
      "max": [
        254,
        254,
        254
      ]
    }
```